### PR TITLE
[MRG+1] [DOC] Turn ginput dostring into a numpydocstring

### PR DESCRIPTION
**Pull request by dstansby**

### CODE REVIEW

File: lib/matplotlib/figure.py [hunk 1]
```diff
@@ -1827,24 +1827,44 @@ def subplots_adjust(self, *args, **kwargs):
...
         points, terminating the inputs) can be overriden via the
         arguments *mouse_add*, *mouse_pop* and *mouse_stop*, that give
         the associated mouse button: 1 for left, 2 for middle, 3 for
         right.
 
+        Parameters
+        ----------
+        n : int, optional
+            Number of mouse clicks to accumulate. If negative, accumulate
+            clicks until the input is terminated manually.
```
Comment (afvincent) [inline lib/matplotlib/figure.py]: Add "Default is 1."?

File: lib/matplotlib/figure.py [hunk 2]
```diff
@@ -1827,24 +1827,44 @@ def subplots_adjust(self, *args, **kwargs):
...
-        This will wait for *n* clicks from the user and return a list of the
-        coordinates of each click.
-
-        If *timeout* is zero or negative, does not timeout.
-
-        If *n* is zero or negative, accumulate clicks until a middle click
-        (or potentially both mouse buttons at once) terminates the input.
-
-        Right clicking cancels last input.
+        Blocking call to interact with a figure. This will wait for *n*
```
Comment (anntzer) [inline lib/matplotlib/figure.py]: I would rewrite the paragraph as
> Wait until the user clicks *n* times on the figure.  Returns the coordinates of each click in a list.

File: lib/matplotlib/figure.py [hunk 3]
```diff
@@ -1827,24 +1827,44 @@ def subplots_adjust(self, *args, **kwargs):
...
         points, terminating the inputs) can be overriden via the
         arguments *mouse_add*, *mouse_pop* and *mouse_stop*, that give
         the associated mouse button: 1 for left, 2 for middle, 3 for
         right.
 
+        Parameters
+        ----------
+        n : int, optional
+            Number of mouse clicks to accumulate. If negative, accumulate
+            clicks until the input is terminated manually. Default is 1.
```
Comment (NelleV) [inline lib/matplotlib/figure.py]: the numpydoc formats puts the default on the first line:

```
n : int, optional, default: 1
```

I find the numpydoc "offical" version more readable.

File: lib/matplotlib/figure.py (hunk 2 above)
Comment (NelleV) [inline lib/matplotlib/figure.py]: Also, I believe (though it is very unsure) that we decided to drop the "*", which are a remainer of the past in Matplotlib's documentation.

Comment (anntzer) [inline lib/matplotlib/figure.py]: Not for *arguments*, see http://matplotlib.org/devdocs/devel/documenting_mpl.html#formatting.

File: lib/matplotlib/figure.py [hunk 4]
```diff
@@ -1827,24 +1827,42 @@ def subplots_adjust(self, *args, **kwargs):
...
+            Mouse button used to add points
+        mouse_pop : int, one of (1, 2, 3), optional, default: 3 (right click)
+            Mouse button used to remove the most recently added point
+        mouse_stop : int, one of (1, 2, 3), optional, default: 2 (middle click)
+            Mouse button used to stop input
+
+        Returns
+        -------
+        points : list of tuples
+            The clicked points. ``len(points)`` will equal the number of points
```
Comment (anntzer) [inline lib/matplotlib/figure.py]: I would rewrite the entire thing (the three sentences) as `A list of (x, y) coordinates of the clicks.`  There is no "number of requested points" if n<0, and there can be fewer points if the user presses `mouse_stop` prematurely.

Comment (dstansby) [inline lib/matplotlib/figure.py]: Yep, sounds better.

File: lib/matplotlib/figure.py [hunk 5]
```diff
@@ -1827,24 +1827,40 @@ def subplots_adjust(self, *args, **kwargs):
...
-        This will wait for *n* clicks from the user and return a list of the
-        coordinates of each click.
-
-        If *timeout* is zero or negative, does not timeout.
-
-        If *n* is zero or negative, accumulate clicks until a middle click
-        (or potentially both mouse buttons at once) terminates the input.
-
-        Right clicking cancels last input.
+        Blocking call to interact with a figure. Wait until the user clicks *n*
```
Comment (QuLogic) [inline lib/matplotlib/figure.py]: The first line should be by itself with the remaining description on another paragraph (one blank line later).

File: lib/matplotlib/figure.py [hunk 6]
```diff
@@ -1827,24 +1827,41 @@ def subplots_adjust(self, *args, **kwargs):
...
+        Parameters
+        ----------
+        n : int, optional, default: 1
+            Number of mouse clicks to accumulate. If negative, accumulate
+            clicks until the input is terminated manually.
+        timeout : scalar, optional, default: 30
+            Number of seconds to wait before timing out. If zero or negative
+            will never timeout.
+        show_clicks : bool, optional, default: False
+            If True, show a red cross at the location of each click
```
Comment (anntzer) [inline lib/matplotlib/figure.py]: Add final dots here and below (including returns), and squash?  (otherwise that's a lot of commits :-))

Comment (phobson) [inline lib/matplotlib/figure.py]: In addition to the comment above, I think the format should be, for example

```
show_clicks : bool, optional (default = False)
```

Comment (QuLogic) [inline lib/matplotlib/figure.py]: That format seems much rarer in our codebase than the existing format.

Comment (anntzer) [inline lib/matplotlib/figure.py]: @QuLogic I will approve the PR after this (the dots -- squashing is up to you) is fixed.

Comment (anntzer) [inline lib/matplotlib/figure.py]: The format for defaults should probably added to the documenting matplotlib developer's guide.  Note that numpy itself gives the example
```
Description of parameter `x` (the default is -1, which implies summation
over all axes).
```
(which is different and I dislike)

Comment (QuLogic) [inline lib/matplotlib/figure.py]: I think you meant to tag @dstansby there; I'm fine with this when you are.

**Comment by dstansby** [comment]
I've added in the extra full stops, and given it a squash.

**Comment by NelleV** [comment]
Thanks @dstansby !

**Pull request merged**